### DISCUSSION TITLE
Dynamically setting the tailwind class

### DISCUSSION THREAD

**Discussion by aminBenSlimen** [tailwindlabs/tailwindcss#7084] [Help] [answered]
so i was trying to dynamically set my background color and this happened , although i didn't face this problem at all in the past 3 months i'm working tailwind with angular 
the only difference is i've used the `ng add` instead of `npm install` i'm not sure what's the difference but i think it's a purge problem.

the issue (the class is added in the DOM but not in the CSS) : 

![image](https://user-images.githubusercontent.com/48415880/149641116-41e7d3ef-9133-4580-94f0-79a9cf36484c.png)

the way i'm setting the class (in ngOnInit method ): 

![image](https://user-images.githubusercontent.com/48415880/149641210-23f37ed2-0765-44d9-a69e-badffb1e37a2.png)

Thank you very much .

**Comment by prasannaraja** [comment]
Hi AminBenSlimen

here is an example for you, you can use ngClass.

` <div [ngClass]="{'bg-sky-500': condition1=== true, 'bg-green-600' : condition2 === true, 'bg-amber-500' : condition3 === true }" class="ml-4 text-xs inline-flex items-center font-bold leading-sm uppercase px-3 py-1 bg-sky-500 text-white border">{{something}}</div>`

**Reply by aminBenSlimen** [reply]
but i don't have condition1 , condition2 ...
what i have is a color dynamicaly changing from database so i have to put it this way : 'bg-' + theColorThatCanBeAnyThing + '-500'

i've got a solution tho , it seems that purgeCss doesn't like it when you use javascript/typescript to set your class , so you can tell the purgecss to not look for specific classes using 'safeLists' like [so](https://tailwindcss.com/docs/content-configuration#safelisting-classes)

[see more](https://stackoverflow.com/questions/70734425/tailwind-angular-does-not-accept-class-names-generated-by-concatenation?noredirect=1#comment125048014_70734425)

**Comment by aminBenSlimen** [comment, marked as answer]
SOLVED ( [solution](https://github.com/tailwindlabs/tailwindcss/discussions/7084#discussioncomment-1981671) )
